optados/src/photo.f90

```diff
@@ -0,0 +1,2722 @@
+            end do
+         end do
+      end do
+
+      qe_norm = sum(qe_temp(1:nbands,1:num_kpoints_on_node(my_node_id),1:nspins,1:max_atoms+1))&
+      /sum(weighted_temp(1:max_energy,1:num_kpoints_on_node(my_node_id),1:nspins,1:nbands,1:max_atoms+1))
+
+      do N=1,num_kpoints_on_node(my_node_id)   ! Loop over kpoints
+         do N_spin=1,nspins                    ! Loop over spins
+            do n_eigen=1,nbands
+               do e_scale=1,max_energy
+                  do atom=1,max_atoms+1
+                     weighted_temp(e_scale,N,N_spin,n_eigen,atom)=&
+                     weighted_temp(e_scale,N,N_spin,n_eigen,atom)*qe_norm
+                  end do
+               end do
+            end do
+         end do
+      end do
+
+      if(allocated(qe_temp)) then
+         deallocate(qe_temp,stat=ierr)
+         if(ierr/=0) call io_error('Error: photo_deallocate - failed to deallocate kpoint_r_cart')
+      end if
+   end if
+   
+   if (index(photo_model,'1step')>0) then
+     do N=1,num_kpoints_on_node(my_node_id)   ! Loop over kpoints
+        do N_spin=1,nspins                    ! Loop over spins
+           do n_eigen=1,nbands
+!           if(band_energy(n_eigen,N_spin,N).ge.efermi) cycle
+              do e_scale=1,max_energy
+                 do atom=1,max_atoms+1
+                    weighted_temp(e_scale,N,N_spin,n_eigen,atom)=&
+                    transverse_temp(e_scale,N,N_spin,n_eigen)*qe_osm(n_eigen,N,N_spin,atom) 
+                 end do
+              end do
+           end do
+        end do
+     end do
+
+     qe_norm = sum(qe_osm(1:nbands,1:num_kpoints_on_node(my_node_id),1:nspins,1:max_atoms+1))&
+     /sum(weighted_temp(1:max_energy,1:num_kpoints_on_node(my_node_id),1:nspins,1:nbands,1:max_atoms+1))
+
+     do N=1,num_kpoints_on_node(my_node_id)   ! Loop over kpoints
+        do N_spin=1,nspins                    ! Loop over spins
+           do n_eigen=1,nbands
+              do e_scale=1,max_energy
+                 do atom=1,max_atoms+1
+                   weighted_temp(e_scale,N,N_spin,n_eigen,atom)=&
+                   weighted_temp(e_scale,N,N_spin,n_eigen,atom)*qe_norm
+                 end do
+              end do
+           end do
+        end do
+     end do
+
+     if(allocated(qe_osm)) then
+        deallocate(qe_osm,stat=ierr)
+        if(ierr/=0) call io_error('Error: photo_deallocate - failed to deallocate kpoint_r_cart')
+     end if
+  end if
+
+     if(allocated(E_transverse)) then
+        deallocate(E_transverse,stat=ierr)
+        if(ierr/=0) call io_error('Error: photo_deallocate - failed to deallocate kpoint_r_cart')
+     end if
+     if(allocated(transverse_temp)) then
+        deallocate(transverse_temp,stat=ierr)
+        if(ierr/=0) call io_error('Error: photo_deallocate - failed to deallocate kpoint_r_cart')
+     end if
+
+  end subroutine transverse_energy_spread
+
+  subroutine binding_energy_spread
+  ! This subroutine applies a Gaussian broadenning to the binding energy
```

Comment:
This looks like a copy of the function above? Why not make a function that does both ?